metagpt/actions/action.py

```diff
@@ -26,16 +26,25 @@ def __init__(self, name: str = "", context=None, llm: LLM = None):
             llm = LLM()
         self.llm = llm
         self.context = context
-        self.prefix = ""
-        self.profile = ""
-        self.desc = ""
-        self.content = ""
-        self.instruct_content = None
+        self.prefix = ""  # aask*时会加上prefix，作为system_message
+        self.profile = ""  # FIXME: USELESS
+        self.desc = ""  # for skill manager
+        self.nodes = ...
+
+        # Output, useless
+        # self.content = ""
+        # self.instruct_content = None
+        # self.env = None
+
+    # def set_env(self, env):
+    #     self.env = env
 
     def set_prefix(self, prefix, profile):
         """Set prefix for later usage"""
         self.prefix = prefix
         self.profile = profile
+        self.llm.system_prompt = prefix
```

Comment:
Now system_prompt will be modified directly, and it will be done after the role obtains llm. This logic is reasonable, and this line can be deleted (because it is a duplicate).